src/charm.py

```diff
@@ -635,13 +610,19 @@ def _wp_generate_password(length=64):
             characters = string.ascii_letters + "!@#$%^&*()" + "-_ []{}<>~`+=,.;:/?|"
             return "".join(secrets.choice(characters) for _ in range(length))
 
-        return {
+        wp_secrets = {
             field: _wp_generate_password()
             for field in self._wordpress_secret_key_fields()
         }
+        wp_secrets["default_admin_password"] = secrets.token_urlsafe(32)
```

Comment:
For my understanding, the `default_admin_password` secret seems to be generated differently to the other secrets using `secrets.token_urlsafe` instead of the `_wordpress_secret_key_fields` function. What is the reason for that please?

Also, it looks like `_wordpress_secret_key_fields` doesn't generate the `default_admin_password` key, should it? Or is it more appropriate to define it specially here?

Reply:
`default_admin_password` is for humans and others are only for WordPress to use internally so they are a little bit different. All secrets listed in `_wordpress_secret_key_fields` is generate so that is perfectly compatible with the PHP function `wp_salt`. The reason `default_admin_password` doesn't use `wp_salt` is because `wp_salt` contains too much special characters that would be too difficult for human to read.

Reply:
Makes sense, thanks